tools/webpack/blocks.js

```diff
@@ -30,6 +30,7 @@ const blockViewRegex = new RegExp(
 const prefixFunctions = [
 	'build_query_vars_from_query_block',
 	'wp_enqueue_block_support_styles',
+	'wp_theme_element_class_name',
```

Comment:
I would be happier if we wouldn't have to list those functions here. What's the reasoning behind using the function name with `gutenberg_prefix`?

Reply:
That was my suggestion. I probably applied the conclusion from https://github.com/WordPress/gutenberg/pull/43779#issuecomment-1236924960 and the [naming collision](https://github.com/WordPress/wordpress-develop/pull/3206#issuecomment-1242990138) too strongly.

Since we're not dealing with a different version of the same function (as was the case in https://github.com/WordPress/gutenberg/pull/43779, which was then superseded by https://github.com/WordPress/gutenberg/pull/43859), it's probably fine to name it `wp_theme_element_class_name` in Gutenberg (and wrap the definition in a `! function_exists()` check).

Reply:
Aside: @mtias has pointed out to me in Slack that

> The one issue with the function check is that it prevents improvements on the plugin to the core functions

-- which was basically the situation we had in #43859. As we've discussed above, this doesn't currently apply to `wp_theme_element_class_name` -- if we ever decide to modify that function, we can still adopt a prefix-rewriting strategy.

We should probably put some guidelines into writing -- there currently seems to be a bit of confusion around what strategies to apply to avoid collisions (see e.g. https://github.com/WordPress/gutenberg/pull/44100).

Reply:
There are guidelines present in the documentation:
https://github.com/WordPress/gutenberg/tree/trunk/lib#wrap-functions-and-classes-with--function_exists-and--class_exists

Feel free to add more details, so it makes the process less confusing.

I think there is rarely a reason to use the trick with webpack and `prefixFunctions` list. Two existing cases are related to the function param changes in the function definition. Otherwise, you can use all sorts of feature detection to figure out what needs to be polyfilled.